app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java

```diff
@@ -239,11 +261,20 @@ public boolean onBackPressed() {
      */
     private void saveTrip(Trip trip) {
         bSave.setActivated(false);
-        CallbackTask.execute(() -> TripDbHelper.instance().updateList(trip), new CallbackTask.DefaultCallback<Object>() {
+        CallbackTask.execute(() -> {
+            TripDbHelper.instance().updateList(trip);
+            TripDbHelper.instance().addUserLandmarks(openTrip.getId(), addedLandmarks);
+            TripDbHelper.instance().deleteUserLandmarks(openTrip.getId(), removedLandmarks);
+            return null;
+        }, new CallbackTask.DefaultCallback<Void>() {
             @Override
-            public void success(Object o) {
+            public void success(Void o) {
                 super.success(o);
+                //Reset lists which track added and removed cards
+                addedLandmarks.clear();
```

Comment:
Clarified by @artemmikhalitsin: the clear() method needs to be called by lists in order to avoid duplicates when the list is regenerated for view in the landmarks screen.